utils/swift_build_support/swift_build_support/products/swiftdriver.py

```diff
@@ -0,0 +1,72 @@
+# swift_build_support/products/swiftdriver.py -------------------*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+
+import os
+
+from . import cmark
+from . import foundation
+from . import indexstoredb
+from . import libcxx
+from . import libdispatch
+from . import libicu
+from . import llbuild
+from . import llvm
+from . import product
+from . import swift
+from . import xctest
+
+from .. import shell
+from .. import targets
+
+class SwiftDriver(product.Product):
+    @classmethod
+    def product_source_name(cls):
+        return "swift-driver"
+
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    def should_build(self, host_target):
+        return self.args.build_swift_driver
+
+    @classmethod
+    def get_dependencies(cls):
+        return [cmark.CMark,
+                llvm.LLVM,
+                libcxx.LibCXX,
```

Comment:
Is libc++ a strict dependency?  What about non-Darwin targets?

Reply:
I cargo-culted this from SwiftPM's dependencies. @gottesmm do you recall why we have the libc++ dependency here?

Reply:
This is b/c I used the most conservative set of dependencies that build-script had. If you don't need it (or these other projects don't) then I would just remove it.

I think of it as a base part of the just built toolchain like clang/compiler-rt. So I imagine it doesn't hurt anything.

Reply:
IOW, build-script had a linear ordering that never changed. I created the simplest correct dependency chain by (ignoring lldb) making all products depend on all other products before them.